distributed/scheduler.py

```diff
@@ -1169,21 +1153,17 @@ def add_duration(self, action: str, start: float, stop: float) -> None:
         self.prefix.add_duration(action, start, stop)
 
     def add(self, other: TaskState) -> None:
-        super().add(other)
-        self.prefix.add(other)
+        self.states[other.state] += 1
+        self.prefix.states[other.state] += 1
         other.group = self
 
     def add_type(self, typename: str) -> None:
-        super().add_type(typename)
-        self.prefix.add_type(typename)
-
-    def transition(self, old: TaskStateState, new: TaskStateState) -> None:
-        super().transition(old, new)
-        self.prefix.transition(old, new)
+        self._types[typename] += 1
+        self.prefix._types[typename] += 1
 
     def update_nbytes(self, diff: int) -> None:
-        super().update_nbytes(diff)
-        self.prefix.update_nbytes(diff)
```

Comment:
haven't benchmarked this but the function dispatch is more epensive than the computation itself.